src/i2c.rs

```diff
@@ -217,13 +223,26 @@ where
             } else if isr.nackf().bit_is_set() {
                 self.i2c.icr.write(|w| w.nackcf().set_bit());
                 return Err(Error::Nack);
+            } else if isr.txe().bit_is_set() {
+                return Ok(());
             }
-            return Ok(());
         }
     }
 
     fn recv_byte(&self) -> Result<u8, Error> {
-        while self.i2c.isr.read().rxne().bit_is_clear() {}
+        while self.i2c.isr.read().rxne().bit_is_clear() {
+            let isr = self.i2c.isr.read();
+            if isr.berr().bit_is_set() {
```

Comment:
What about reusing the whole error handling block? send_byte uses almost the same thing. It can be separated to the function.

Reply:
This has been done in #102, I think, so rebase should take care of this comment.